pandas/core/categorical.py

```diff
@@ -294,22 +293,10 @@ def __init__(self, values, categories=None, ordered=None, dtype=None,
 
         # sanitize input
         if is_categorical_dtype(values):
+            if dtype.categories is None:
+                dtype = CategoricalDtype(values.categories, dtype.ordered)
```

Comment:
what kind of construction actually hits this?

it is a bit inconsistent, e.g. if values is a Categorical then it determines the categories but NOT the ordered? 
what if dtype differs from the dtype of values?

Reply:
> it is a bit inconsistent, e.g. if values is a Categorical then it determines the categories but NOT the ordered?

Yes, it does... above (``if dtype is not None``). That is: ``dtype`` here is not necessarily the ``dtype`` argument to the function.

> what if dtype differs from the dtype of values?

Then it is given priority - rightly so, I think.

Reply:
ok, can you add a comment to that effect here.

> it is a bit inconsistent, e.g. if values is a Categorical then it determines the categories but NOT the ordered?

is there a case which hits this where the passed dtype has a *different* ordered than ``values.ordered``? (one is False, one is True). do/should we check this?

Reply:
why are we not simply using ``dtype = values`` at this point (``values`` is already a dtype)?

Reply:
> why are we not simply using dtype = values at this point (values is already a dtype)?

``values`` can _have_ a dtype...

> is there a case which hits this where the passed dtype has a different ordered than values.ordered? (one is False, one is True). do/should we check this?

Sure, whenever you pass values with given categories, unordered, and dtype with the same categories, ordered. Anyway, I added a couple of comments.